### PR TITLE
Improve compression of TastyString

### PR DESCRIPTION
**Pull request by nicolasstucki**
Improves #3877

`'{ { val y = ~x * ~x; ~powerCode(n / 2, '(y)) } }`
was reduced from 206 bytes to 172 bytes

`'{ ~x * ~powerCode(n - 1, x) }`
was reduced from 143 bytes to 128 bytes

### CODE REVIEW

File: compiler/src/dotty/tools/dotc/core/tasty/TastyString.scala [hunk 1]
```diff
@@ -2,34 +2,34 @@ package dotty.tools.dotc.core.tasty
...
-  /** Decode the TASTY String into TASTY bytes */
-  def unpickle(strings: Pickled): Array[Byte] = {
-    val bytes = new Array[Byte](strings.map(_.length).sum)
-    var i = 0
-    for (str <- strings; j <- str.indices) {
-      bytes(i) = str.charAt(j).toByte
-      i += 1
+  def pickle(bytes: Array[Byte]): Pickled = {
+    val str = new String(Base64.getEncoder().encode(bytes), UTF_8)
+    def split(sliceEnd: Int, acc: List[String]): List[String] = {
```
[the remark this answers is not in the document]
Comment (nicolasstucki) [reply compiler/src/dotty/tools/dotc/core/tasty/TastyString.scala]: This version is more efficient.

**Comment by nicolasstucki** [comment]
This does not affect the tasty format. We do not need to update the version number.

**Comment by odersky** [comment]
I don't know what the policy wrt bumbing the Tasty version is. Say you have two PR's and they each bump the version from 11 to 12. Then what is the right version? Alternatively, could we bump the TASTY version only for each release if there is a change in TASTY? That would limit the bumping speed to at most one every 6 weeks or so. I am not sure what scheme is preferable. What do others think?

**Comment by nicolasstucki** [comment]
Regardless of the Tasty version bumping scheme, this PR does not need a tasty version bump. Maybe I should add a version to the scheme used to serialize tasty bytes into strings.

File: compiler/src/dotty/tools/dotc/core/tasty/TastyString.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (allanrenucci) [reply compiler/src/dotty/tools/dotc/core/tasty/TastyString.scala]: My personal opinion is to go with the simplest solution unless it is performance sensitive and it happen to be a hotspot. No strong feelings though

[the remark this answers is not in the document]
Comment (nicolasstucki) [reply compiler/src/dotty/tools/dotc/core/tasty/TastyString.scala]: I will change it to the sliding. We can always change it later if needed.

**Pull request merged**